rails_event_store_active_record/lib/rails_event_store_active_record/event_repository.rb

```diff
@@ -146,10 +117,61 @@ def get_all_streams
 
     attr_reader :mapper
 
+    def add_to_stream(collection, stream_name, expected_version, include_global, &to_event_id)
```

Comment:
for me this methods still doing to much, it's difficult to reason about it, please reconsider extract few well named methods from it, to show on this level only top view of the process

Reply:
I found 3 methods which definitely make sense extracting. There is potential for more but I could not find good scope and names.